src/fswAlgorithms/pointCloudProcessing/cameraTriangulation/_UnitTest/test_cameraTriangulation.py

```diff
@@ -0,0 +1,232 @@
+
+    # Configure input messages
+    pointCloudInMsgData = messaging.PointCloudMsgPayload()
+    pointCloudInMsgData.timeTag = time
+    pointCloudInMsgData.valid = True
+    pointCloudInMsgData.numberOfPoints = numberOfPoints
+    pointCloudInMsgData.points = pointCloudStacked
+    pointCloudInMsg = messaging.PointCloudMsg().write(pointCloudInMsgData)
+
+    keyPointsInMsgData = messaging.PairedKeyPointsMsgPayload()
+    keyPointsInMsgData.valid = True
+    keyPointsInMsgData.cameraID = ID
+    keyPointsInMsgData.timeTag_secondImage = time
+    keyPointsInMsgData.keyPoints_secondImage = keyPointsStacked
+    keyPointsInMsgData.sigma_BN_secondImage = sigma_BN
+    keyPointsInMsgData.keyPointsFound = numberOfPoints
+    keyPointsInMsg = messaging.PairedKeyPointsMsg().write(keyPointsInMsgData)
+
+    cameraConfigInMsgData = messaging.CameraConfigMsgPayload()
+    cameraConfigInMsgData.cameraID = ID
+    cameraConfigInMsgData.fieldOfView = fov
+    cameraConfigInMsgData.resolution = res
+    cameraConfigInMsgData.cameraPos_B = r_CB_B
+    cameraConfigInMsgData.sigma_CB = sigma_CB
+    cameraConfigInMsg = messaging.CameraConfigMsg().write(cameraConfigInMsgData)
+
+    # subscribe input messages to module
+    module.pointCloudInMsg.subscribeTo(pointCloudInMsg)
+    module.keyPointsInMsg.subscribeTo(keyPointsInMsg)
+    module.cameraConfigInMsg.subscribeTo(cameraConfigInMsg)
+
+    # setup output message recorder objects
+    cameraLocationOutMsgRec = module.cameraLocationOutMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, cameraLocationOutMsgRec)
+
+    unitTestSim.InitializeSimulation()
+    unitTestSim.TotalSim.SingleStepProcesses()
+
+    # pull module data
+    cameraLocation = cameraLocationOutMsgRec.cameraPos_N[0]
+    valid = cameraLocationOutMsgRec.valid[0]
+
+    # true data
+    cameraLocationTrue = r_CN_N
+    if (pointCloudInMsgData.valid and
+    keyPointsInMsgData.valid and
+    keyPointsInMsgData.cameraID == cameraConfigInMsgData.cameraID and
+    pointCloudInMsgData.timeTag == keyPointsInMsgData.timeTag_secondImage and
+    pointCloudInMsgData.numberOfPoints == keyPointsInMsgData.keyPointsFound):
+        validTrue = True
+    else:
+        validTrue = False
+
+    # make sure module output data is correct
+    paramsString = ' for true anomaly={}, camera position={}, SC rotation={}, accuracy={}'.format(
+        str(p1_f),
+        str(p2_cam),
+        str(p3_scRot),
+        str(accuracy))
+
+    np.testing.assert_allclose(cameraLocation,
+                               cameraLocationTrue,
+                               rtol=0,
+                               atol=accuracy,
+                               err_msg=('Variable: cameraLocation,' + paramsString),
+                               verbose=True)
+
+    # np.testing.assert_equal(valid,
+    #                         validTrue,
+    #                         err_msg=('Variable: valid flag,' + paramsString),
+    #                         verbose=True)
```

Comment:
Waiting for the swigging of bools to work?

Reply:
Yes.